File: backend/app/utils/phone.py
import re


def normalize_phone(phone: str) -> str:
    """
    Normalisasi nomor HP Indonesia ke format 628xxxx.
    Mendukung: 08xxx, +628xxx, 628xxx, 8xxx
    """
    phone = re.sub(r"[\s\-\(\)]", "", phone)
    if phone.startswith("+62"):
        return phone[1:]
    elif phone.startswith("62"):
        return phone
    elif phone.startswith("08"):
        return "62" + phone[1:]
    elif phone.startswith("8"):
        return "62" + phone
    return phone


def to_wa_link(phone: str) -> str:
    """Konversi nomor HP ke link WhatsApp."""
    normalized = normalize_phone(phone)
    return f"https://wa.me/{normalized}"


def to_call_link(phone: str) -> str:
    """Konversi nomor HP ke link telepon."""
    normalized = normalize_phone(phone)
    return f"tel:+{normalized}"

File: frontend/lib/phone.ts
export function normalizePhone(phone: string): string {
  const cleaned = phone.replace(/[\s\-\(\)]/g, '');
  if (cleaned.startsWith('+62')) return cleaned.slice(1);
  if (cleaned.startsWith('62')) return cleaned;
  if (cleaned.startsWith('08')) return '62' + cleaned.slice(1);
  if (cleaned.startsWith('8')) return '62' + cleaned;
  return cleaned;
}

export function toWaLink(phone: string): string {
  return `https://wa.me/${normalizePhone(phone)}`;
}

export function toCallLink(phone: string): string {
  return `tel:+${normalizePhone(phone)}`;
}
